tests/src/python/test_qgsdelimitedtextprovider_wanted.py

```diff
@@ -893,11 +893,11 @@ def test_018_regular_expression_2():
             'GEXP': 'GEXP',
             'description': 'RE',
             'RE_1': 'RE',
-            'GEXP_1': 'NULL',
-            'data': 'data2',
-            'RE_2': 'RE',
+            'GEXP_1': 'data2',
+            'data': 'RE',
+            'RE_2': 'info2',
             'GEXP_2': 'NULL',
-            'info': 'info2',
+            'info': 'NULL',
```

Comment:
Was this test wrong or did the actual behavior change?

Reply:
@uclaros , QRegularExpression behavior changed (to match other regular expression engines).

That test's regular expression was (RE)(GEXP)?, where the second group is optional, it can either be present once or not. 

With QRegExp, an empty optional group would be present in capturedTexts(), while QRegularExpression (correctly) doesn't add it.

Reply:
I could have updated the regular expression instead to make more sense in this context: 

i.e., (RE)((GEXP)?) instead of (RE)(GEXP)?